lighthouse-core/test/fraggle-rock/scenarios/stringified-replay-test-pptr.js

```diff
@@ -0,0 +1,127 @@
+ */
+
+import fs from 'fs';
+import {promisify} from 'util';
+import {execFile} from 'child_process';
+
+import {stringify} from '@puppeteer/replay';
+
+import {LH_ROOT} from '../../../../root.js';
+import LighthouseStringifyExtension from '../../../fraggle-rock/replay/stringify-extension.js';
+import {getAuditsBreakdown, createTestState} from './pptr-test-utils.js';
+import {readJson} from '../../test-utils.js';
+
+const execFileAsync = promisify(execFile);
+const desktopReplayJson = readJson('lighthouse-core/test/fixtures/fraggle-rock/replay/desktop-test-flow.json');
+const mobileReplayJson = readJson('lighthouse-core/test/fixtures/fraggle-rock/replay/mobile-test-flow.json');
+const FLOW_JSON_REGEX = /window\.__LIGHTHOUSE_FLOW_JSON__ = (.*);<\/script>/;
+
+describe('Running the stringified output script', function() {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(60_000);
+
+  // Flow JSON specifies port 10200 so we have to use that for the server.
+  const state = createTestState();
+  state.installServerHooks(10200);
+
+  const tmpDir = `${LH_ROOT}/.tmp/replay`;
+  let testTmpDir = '';
+  let scriptPath = '';
+
+  before(async () => {
+    fs.mkdirSync(tmpDir, {recursive: true});
+  });
+
+  beforeEach(async () => {
+    testTmpDir = fs.mkdtempSync(`${tmpDir}/replay-`);
+    scriptPath = `${testTmpDir}/stringified.cjs`;
+  });
+
+  after(async () => {
+    fs.rmSync(tmpDir, {recursive: true, force: true});
+  });
+
+  it('generates a valid desktop flow report', async () => {
+    const scriptContents = await stringify(desktopReplayJson, {
+      extension: new LighthouseStringifyExtension(),
+    });
+
+    expect(scriptContents).toMatchSnapshot();
+    fs.writeFileSync(scriptPath, scriptContents);
+
+    const {stdout, stderr} = await execFileAsync('node', [scriptPath], {timeout: 50_000});
+
+    // Ensure script didn't quietly report an issue.
+    expect(stdout).toEqual('');
+    expect(stderr).toEqual('');
+
+    const reportHtml = fs.readFileSync(`${testTmpDir}/flow.report.html`, 'utf-8');
+    const flowResultJson = FLOW_JSON_REGEX.exec(reportHtml)?.[1];
+    if (!flowResultJson) throw new Error('Could not find flow json');
+
+    /** @type {LH.FlowResult} */
+    const flowResult = JSON.parse(flowResultJson);
+    expect(flowResult.name).toEqual(desktopReplayJson.title);
+    expect(flowResult.steps.map(step => step.lhr.gatherMode)).toEqual([
+      'navigation',
+      'timespan',
+      'navigation',
+      'timespan',
+    ]);
+
+    for (const {lhr} of flowResult.steps) {
+      expect(lhr.configSettings.formFactor).toEqual('desktop');
+      expect(lhr.configSettings.screenEmulation.disabled).toBeTruthy();
+
+      const {auditResults, erroredAudits} = getAuditsBreakdown(lhr);
+      expect(auditResults.length).toBeGreaterThanOrEqual(10);
+
+      // TODO: INP breakdown diagnostic audit is broken because of old Chrome
```

Comment:
out of date now?

Reply:
The stringified script will use the Chrome shipped with Puppeteer, so we need to update our Puppeteer to fix this.